app/controllers/nonsipp/employercontributions/package.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp
+
+import config.Refined.{Max300, Max50}
+import models.SchemeId.Srn
+import models.UserAnswers
+import pages.nonsipp.employercontributions.EmployerContributionsProgress
+import play.api.mvc.Call
+import utils.FunctionKUtils._
+import viewmodels.models.SectionJourneyStatus
+
+import scala.concurrent.Future
+
+package object employercontributions {
+
+  implicit class CallOps(call: Call) {
+
+    val isCyaPage: Boolean = {
+      val pattern =
+        s"^\\/pension-scheme-return\\/[^\\/]+\\/(change|check-answers)-employer-contributions\\/(?!.*\\/\\d+\\/\\d+$$).*"
```

Comment:
Is there any risk that this is not going to work in the cloud environments? In case there are any issues with this we may need to choose some different way to identify the CYA page.

Reply:
I'm matching using the `url` value of the `Call` object which removes the host so should be fine